test/Microsoft.ML.TestFramework/DataPipe/TestDataPipe.cs

```diff
@@ -1300,6 +1302,49 @@ public void ArrayDataViewBuilderNoCols()
             Done();
         }
 
+        [Fact]
+        public void TestBatchTransform()
+        {
+            // TODO: delete
+            var bldr = new ArrayDataViewBuilder(Env);
+            bldr.AddColumn("Input", NumberDataViewType.Single, new[] { 1f, 2f, 3f, 2f, 3f, 4f, 3f, 4f, 5f, 4f });
+            var input = bldr.GetDataView();
+
+            var output = new DetectAnomalyBySrCnnBatchTransform(
+                Env, 
+                input, 
+                "Input", 
+                "Output",
+                0.3,
+                3,
+                99,
+                DetectAnomalyBySrCnnBatchTransform.DetectMode.AnomalyAndExpectedValue);
+            // var batchTransformOutput = ML.Data.CreateEnumerable<BatchTransformOutput>(output, reuseRowObject: false).ToList();
```

Comment:
This does not work. CreateEnumerable requires a vector output per the schema, but the actual output is a flow. Is there a way to make CreateEnumerable work in this scenario?

Reply:
Can you please elaborate? I am not sure I understand what you mean by flow. 

---
In reply to: [424168522](https://github.com/mengaims/machinelearning/pull/1#discussion_r424168522) [](ancestors = 424168522)

Reply:
Sorry "float"

Reply:
Hi @klausmh, the reason is that the getter of the sample class (defined in DetectAnomalyBySrCnnBatchTransform line 318) actually returns a float. For some reason I defined the type of the output column to be a vector of double (line 222), but it should match the type in the getter.

---
In reply to: [424637321](https://github.com/mengaims/machinelearning/pull/1#discussion_r424637321) [](ancestors = 424637321)